framework/src/engine/legacy/legacy_chain_handler.ts

```diff
@@ -31,9 +32,19 @@ export class LegacyChainHandler {
 		this._legacyConfig = args.legacyConfig;
 	}
 
-	// eslint-disable-next-line @typescript-eslint/require-await
 	public async init(args: LegacyHandlerInitArgs): Promise<void> {
+		const storage = new Storage(this._db as Database);
 		this._db = args.db;
+
+		for (const bracket of this._legacyConfig.brackets) {
+			if (!bracket.snapshotBlockID) {
```

Comment:
you also need to check `storage.getLegacyChainBracketInfo(bracket.snapshotBlockID)` exists or not, we only specify defaults when it doesn't exist in DB